datastore/src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -172,6 +173,26 @@
         return loadEntitySetData( entitySetId, selection );
     }
 
+    @Override
+    @RequestMapping(
+            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + FILTERED },
+            method = RequestMethod.POST,
+            consumes = { MediaType.APPLICATION_JSON_VALUE },
+            produces = { MediaType.APPLICATION_JSON_VALUE }
+    )
+    public Iterable<Map<FullQualifiedName, Set<Object>>> loadFilteredEntitySetData(
+            @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
+            @RequestBody FilteredDataPageDefinition filteredDataPageDefinition
+    ) {
+        ensureReadAccess( new AclKey( entitySetId ) );
+        ensureReadAccess( new AclKey( entitySetId, filteredDataPageDefinition.getPropertyTypeId() ) );
+
+        Map<UUID, Map<UUID, PropertyType>> authorizedPropertyTypes = authzHelper
+                .getAuthorizedPropertiesOnEntitySets( ImmutableSet.of( entitySetId ), EnumSet.of( Permission.READ ) );
```

Comment:
`getAuthorizedPropertiesOnEntitySets` will give you all the property types so what does `filteredDataPageDefinition.getPropertyTypeId()` represent?